storage/testing/storeonce.go

```diff
@@ -66,3 +68,30 @@ func (ts *Storage) Store(ctx context.Context, msg nodeenrollment.MessageWithId)
 
 	return ts.Storage.Store(ctx, msg)
 }
+
+// LoadByNodeId implements the NodeIdLoader interface. Iterate through all NodeInformation records
+// and return those matching the NodeId
+func (ts *Storage) LoadByNodeId(ctx context.Context, msg nodeenrollment.MessageWithNodeId) error {
+
+	switch t := msg.(type) {
+	case *types.NodeInformations:
+		nodes, err := ts.Storage.List(ctx, (*types.NodeInformation)(nil))
+		if err != nil {
+			return err
+		}
+
+		nodesToReturn := make([]*types.NodeInformation, 0)
+		for _, n := range nodes {
+			node := &types.NodeInformation{Id: n}
+			ts.Load(ctx, node)
+			if node.NodeId == msg.GetNodeId() {
```

Comment:
Should we check that `msg.NodeId` is `!= ""` to avoid fetching two empty matching nodes?

Reply:
sure!